### ISSUE TITLE
Return ErrorCode with namespace in macros

### ISSUE THREAD

**Issue by yecol** [alibaba/libvineyard#106]
**Describe your problem**

- I occurred an error on macOS when building graphscope. error messages are shown as below:
```
/Users/yecol/git-repo/graphscope-edu/analytical_engine/core/fragment/arrow_projected_fragment.h:891:7: error: use of undeclared identifier
      'ErrorCode'; did you mean 'vineyard::ErrorCode'?
      ARROW_OK_OR_RAISE(begins_builder.Append(ret.first));
      ^
/usr/local/include/vineyard/graph/utils/error.h:220:23: note: expanded from macro 'ARROW_OK_OR_RAISE'
      RETURN_GS_ERROR(ErrorCode::kArrowError, (status_name).ToString()); \
                      ^
/usr/local/include/vineyard/graph/utils/error.h:31:12: note: 'vineyard::ErrorCode' declared here
enum class ErrorCode {
           ^
In file included from /Users/yecol/git-repo/graphscope-edu/analytical_engine/core/grape_instance.cc:20:
In file included from /Users/yecol/git-repo/graphscope-edu/analytical_engine/core/context/tensor_context.h:37:
/Users/yecol/git-repo/graphscope-edu/analytical_engine/core/fragment/arrow_projected_fragment.h:892:7: error: use of undeclared identifier
      'ErrorCode'; did you mean 'vineyard::ErrorCode'?
      ARROW_OK_OR_RAISE(ends_builder.Append(ret.second));
      ^
/usr/local/include/vineyard/graph/utils/error.h:220:23: note: expanded from macro 'ARROW_OK_OR_RAISE'
      RETURN_GS_ERROR(ErrorCode::kArrowError, (status_name).ToString()); \
```

I suggest we return the errorcode enum class together with namespace. 
Since libvineyard may be used by another project which is on the outside of namespace vineyard. Just like graphscope.

- btw, I also suggest an inprovement on find_vineyard, give a hint like other libraries:
```
NOT_Found libvineyard
# or 
Found libvineyard, (include: /usr/local/include, library: /usr/local/lib/libvineyard.dylib)
```

**Descirption of the problem**

If it is a bug report, to help us reproducing this bug, please provide information below:

1. Your Operation System version (`uname -a`): macOS
2. The version of libvineyard you use (`vineyard.__version__`): latest
3. Versions of crucial packages, such as mpi:
4. Full stack of the error: aforementioned
5. Minimized code to reproduce the error: cmake && make -j4 on macOS

If it is a feature request, please provides a clear and concise description of what you want to happen:

1. What is the problem:
2. The behaviour that you expect to work:

**Additional context**

**Comment by sighingnow** [comment]
Thanks for reporting!

**Comment by sighingnow** [comment]
Looks that you are working with a older version of vineyard. I will enforce the version requirement in `find_package(vineyard)` in graphscope.

And thanks for the suggestion of the "Found libvineyard" hint.

**Comment by sighingnow** [comment]
The "Found vineyard" message part has been fixed by ccac1d6c76b. We use `find_package(vineyard REQUIRED)` in graphscope that there will be a cmake error if vineyard not found.

The version mismatch issue will be fixed by alibaba/GraphScope#39, which enforce vineyard>=0.1.3 requirement.
